src/test/common/Throttle.cc

```diff
@@ -58,6 +59,8 @@ class ThrottleTest : public ::testing::Test {
     }
   };
 
+  static void destruct_throttle() {
```

Comment:
@adamemerson what is this method for? did you plan to extract the common piece into it and reuse it?